airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py

```diff
@@ -0,0 +1,370 @@
+        expected_params = (0, 2)
+        inputs = {"response": response}
+        get_backoff_time = stream.backoff_time(**inputs)
+        assert expected_params[0] <= get_backoff_time <= expected_params[1]
+
+    def test_okta_stream_incremental_get_updated_state(self, patch_base_class, latest_record_instance, url_base):
+        class TestIncrementalOktaStream(IncrementalOktaStream, ABC):
+            def __init__(self, url_base: str, *args, **kwargs):
+                super().__init__(url_base, *args, **kwargs)
+                self._cursor_field = None
+
+            @property
+            def cursor_field(self) -> str:
+                return self._cursor_field
+
+        stream = TestIncrementalOktaStream(url_base=url_base)
+        stream._cursor_field = "lastUpdated"
+
+        current_stream_state = {"lastUpdated": "2021-04-21T21:03:55.000Z"}
+        update_state = stream.get_updated_state(current_stream_state=current_stream_state, latest_record=latest_record_instance)
+        expected_result = {"lastUpdated": "2022-07-18T07:58:11.000Z"}
+        assert update_state == expected_result
+
+    def test_okta_stream_http_method(self, patch_base_class, url_base):
+        stream = OktaStream(url_base=url_base)
+        expected_method = "GET"
+        assert stream.http_method == expected_method
+
+
+class TestNextPageToken:
+
+    def test_next_page_token(self, patch_base_class, users_instance, url_base, api_url):
+        stream = OktaStream(url_base=url_base)
+        response = MagicMock(requests.Response)
+        response.links = {"next": {"url": f"{api_url}?param1=test_value1&param2=test_value2"}}
+        inputs = {"response": response}
+        expected_token = {"param1": "test_value1", "param2": "test_value2"}
+        result = stream.next_page_token(**inputs)
+        assert result == expected_token
+
+    def test_next_page_token_empty_params(self, patch_base_class, users_instance, url_base, api_url):
+        stream = OktaStream(url_base=url_base)
+        response = MagicMock(requests.Response)
+        response.links = {"next": {"url": f"{api_url}"}}
+        inputs = {"response": response}
+        expected_token = {}
+        result = stream.next_page_token(**inputs)
+        assert result == expected_token
+
+    def test_next_page_token_link_have_self_and_equal_next(self, patch_base_class, users_instance, url_base, api_url):
+        stream = OktaStream(url_base=url_base)
+        response = MagicMock(requests.Response)
+        response.links = {"next": {"url": f"{api_url}"}, "self": {"url": f"{api_url}"}}
+        inputs = {"response": response}
+        expected_token = None
+        result = stream.next_page_token(**inputs)
+        assert result == expected_token
+
+
+class TestStreamUsers:
+
+    def test_stream_users(self, requests_mock, patch_base_class, users_instance, url_base, api_url):
+        stream = Users(url_base=url_base)
+        record = users_instance
```

Comment:
I'm not sure that we need resign here variable. 

Check please other tests.

Reply:
changed in others too
done